File: app/src/main/java/com/example/plantea/dominio/Pictograma.java
package com.example.plantea.dominio;

import android.app.Activity;

import java.io.Serializable;
import java.util.ArrayList;

public class Pictograma implements Serializable {

    private String titulo;
    private String imagen;
    private int categoria;
    private int cuaderno;
    ArrayList<Pictograma> listaPictogramas;
    ArrayList<String> listaConsultas;
    GestionPictogramas gestorPictogramas = new GestionPictogramas();

    public Pictograma(){

    }

    public Pictograma(String titulo, String imagen, int categoria, int cuaderno) {
        this.titulo = titulo;
        this.imagen = imagen;
        this.categoria = categoria;
        this.cuaderno= cuaderno;
    }

    public String getTitulo() {
        return titulo;
    }

    public void setTitulo(String titulo) {
        this.titulo = titulo;
    }

    public String getImagen() {
        return imagen;
    }

    public void setImagen(String imagen) {
        this.imagen = imagen;
    }

    public int getCategoria(){
        return categoria;
    }

    public void setCategoria(int categoria){
        this.categoria = categoria;
    }

    public int getCuaderno(){
        return cuaderno;
    }

    public void setCuaderno(int cuaderno){
        this.cuaderno = cuaderno;
    }

    public ArrayList obtenerPictogramas(Activity actividad, int idcategoria ) {
        listaPictogramas = new ArrayList<>();
        listaPictogramas = gestorPictogramas.listarPictogramas(actividad, idcategoria);
        return listaPictogramas;
    }

    public void nuevoPictograma (Activity actividad, String nombre, String imagen, String categoria){
        gestorPictogramas.insertarPictograma(actividad, nombre, imagen, categoria);
    }

    public ArrayList obtenerPictogramasCuaderno(Activity actividad, int identificador ) {
        listaPictogramas = new ArrayList<>();
        listaPictogramas = gestorPictogramas.listarPictogramasCuaderno(actividad, identificador);
        return listaPictogramas;
    }

    public ArrayList obtenerConsultas(Activity actividad, int idcategoria ) {
        listaConsultas = new ArrayList<>();
        listaConsultas = gestorPictogramas.listarConsultas(actividad, idcategoria);
        return listaConsultas;
    }

    public String obtenerImagenEvento(Activity actividad, String consulta, int idCategoria){
        String ruta;
        ruta = gestorPictogramas.obtenerImagenPictograma(actividad, consulta, idCategoria);
        return ruta;
    }
}

File: app/src/main/java/com/example/plantea/dominio/Pictograma.kt
package com.example.plantea.dominio

import android.app.Activity
import android.content.Context
import android.graphics.Bitmap
import java.io.Serializable

class Pictograma: Serializable {
    var id: String? = null
    var titulo: String? = null

    @Transient
    var imagen: Bitmap? = null

    var idAPI = 0
    var categoria = 0
    var historia: String? = null
    var duracion: String? = null
    var pictoEntretenimiento = 0
    var favorito: Boolean = false
    var listaPictogramas: ArrayList<Pictograma>? = null
    private var gestorPictogramas = GestionPictogramas()

    constructor()
    constructor(titulo: String?, imagen: Bitmap?, idAPI : Int, categoria: Int) {
        this.titulo = titulo
        this.imagen = imagen
        this.categoria = categoria
        this.idAPI = idAPI
    }

    constructor(id:String?, titulo: String?, imagen: Bitmap?, idAPI: Int, categoria: Int, favorito: Boolean) {
        this.id = id
        this.titulo = titulo
        this.imagen = imagen
        this.idAPI = idAPI
        this.categoria = categoria
        this.favorito = favorito
    }

    fun obtenerPictogramas(context: Context, idcategoria: Int, idUsuario: String?, language: String): ArrayList<*> {
        return gestorPictogramas.obtenerPictogramas(context, idcategoria, idUsuario, language)
    }

    fun obtenerFavoritos(actividad: Context?, idUsuario: String?, language: String): ArrayList<Pictograma> {
        listaPictogramas = ArrayList()
        listaPictogramas = gestorPictogramas.obtenerFavoritos(actividad, idUsuario, language)

        return listaPictogramas!!
    }

    fun nuevoPictogramaLocal(actividad: Activity?, nombre: String?, imagen: ByteArray?, categoria: String?, idUsuario: String?): String {
       return gestorPictogramas.insertarPictogramaLocal(actividad, nombre, imagen, categoria, idUsuario)
    }

    fun nuevoPictogramaAPI(actividad: Activity?, nombre: String?, id: String?, categoria: String?): String {
        return gestorPictogramas.insertarPictogramaAPI(actividad, nombre, id, categoria)
    }
    fun insertarFavorito(context: Context?, idUsuario: String?, id: String?, titulo: String?, idAPI: Int) {
        gestorPictogramas.insertarFavorito(context, idUsuario, id, titulo, idAPI)
    }

    fun borrarFavorito(context: Context?, idUsuario: String?, idPicto: String?) {
        gestorPictogramas.borrarFavorito(context, idUsuario, idPicto)
    }

    fun getFavorito(actividad: Activity?, idPicto: String?, idUsuario: String?): Boolean {
        return gestorPictogramas.getFavorito(actividad, idPicto, idUsuario)
    }

    fun obtenerPicto(context: Context?, id: String?, language: String): Pictograma {
        return gestorPictogramas.obtenerPicto(context, id, language)
    }

    fun getRandomPictograms(context: Context?, idUsuario: String?, language: String): ArrayList<Pictograma> {
        return gestorPictogramas.getRandomPictograms(context, idUsuario, language)
    }

    fun copy(): Pictograma {
        return Pictograma(this.id, this.titulo, this.imagen, this.idAPI, this.categoria, this.favorito)
    }

    fun guardarHistoria(actividad: Activity?, id: String, idPicto: String?, historia: String?) {
        gestorPictogramas.guardarHistoria(actividad, id, idPicto, historia)

    }

    fun guardarDuracion(actividad: Activity?, toString: String, id: String, s: String?) {
        gestorPictogramas.guardarDuracion(actividad, toString, id, s)
    }

    fun guardarPictoEntretenimiento(actividad: Activity?, id: String, idPicto: String?, idEntretenimiento: String?) {
        gestorPictogramas.guardarPictoEntretenimiento(actividad, id, idPicto, idEntretenimiento)
    }

}